10up/block-components
# New component: Video

### Description of the Change


Adds a `Video` component to match [`Image`](), and a `VideoControl` component to simplify uploading and other aspects of the video.

Adds **incomplete** documentation for the new components.

Outstanding questions:

- [ ] Should this include (optional) controls for managing `poster` images?
- [ ] Should this include (optional) controls for uploading the semi-mandatory subtitle `<track>`? If so, this may need to be paired with PHP code allowing `vtt` files to be uploaded to WordPress. WordPress' core Video Block allows subtitles; we should borrow from their subtitle components where possible. https://wordpress.org/documentation/article/video-block/
- [ ] Should this include controls for `autoplay`, `loop`, `muted`, `controls`, and `playsinline`?
- [ ] Should this include (optional) controls for different video tracks in different resolutions? Or would a "different videos at different viewports" component be better done with a higher-level "multiple video" component with multiple `VideoControls`?
- [ ] Should this allow the programmer to list an array  `allowedTypes` of specific MIME types that are allowed for upload, instead of defaulting to the wide-open `"video"` or the more-limited `[ 'video/webm', 'video/mp4']`?


Closes #220

### How to test the Change

Testing suggestions are welcome.

- add the `Video` and `VideoControl` components to a block; make sure they work

### Changelog Entry

> Added - New feature
> Changed - Existing functionality
> Deprecated - Soon-to-be removed feature
> Removed - Feature
> Fixed - Bug fix
> Security - Vulnerability


### Credits

Props @username, @username2, ...


### Checklist:
